### PR TITLE
Support more acceptance tests from ./assess helper

### PR DESCRIPTION
**Pull request by timClicks**
## Description of change

Juju's acceptance tests use several variants in their command-line arguments. This change teaches the `acceptancetests/assess` helper script about almost all of them.  `assess_heterogenous_control.py` is particularly idiosyncratic and has been deferred to a future commit.

## QA steps

n/a (developer-facing only)

## Documentation changes

n/a (developer-facing only)

## Bug reference

None.

### CODE REVIEW

File: acceptancetests/assess [hunk 1]
```diff
@@ -162,14 +166,114 @@ def parse_args():
...
+                              help="Path to directory with pre-loaded metadata. (Revelant only for the bootstrap test)")
+    pass_through.add_argument("--remote-metadata-source",
+                              help="URL to image metadata (Relevant only for the mixed_images test)")
+    pass_through.add_argument("--export-to",
+                              help="Part to where an exported bundle will be written to. (Relevant only for the bundle_export test)")
+    pass_through.add_argument("--caas-provider",
+                              help="Provider for CaaS models. (Relevant only for caas_deploy_charms test) [default: microk8s]",
+                              choices=["microk8s", "k8s"])
+    pass_through.add_argument('--caas-image-repo',
+                              help="Docker image repo to use. (Relevant only for caas_deploy_charms test) [default: jujuqabot]",
```
Comment (wallyworld) [inline acceptancetests/assess]: This is relevant for any caas test. As is --caas-provider etc. We only have one test now but are adding more. Can we reword to be more generic.
Not sure if there's other examples also that are currently a little too specific.

File: acceptancetests/assess [hunk 2]
```diff
@@ -263,14 +381,151 @@ def main():
...
+        testrun_argv += ["--to", args.to]
     if args.agent_url:
-        testrun_argv.extend(["--agent-url", args.agent_url])
+        testrun_argv += ["--agent-url", args.agent_url]
     if args.agent_stream:
-        testrun_argv.extend(["--agent-stream", args.agent_stream])
-
+        testrun_argv += ["--agent-stream", args.agent_stream]
+
+    assess = args.assess
```
Comment (wallyworld) [inline acceptancetests/assess]: Can we add a big comment that the below test specific behaviour is only for tests which use bespoke args for that test; tests which just use generic args are not listed etc etc

File: acceptancetests/assess (hunk 1 above)
Comment (timClicks) [reply acceptancetests/assess]: Have created a new section.

File: acceptancetests/assess (hunk 2 above)
Comment (timClicks) [reply acceptancetests/assess]: Sure, done.

**Comment by timClicks** [comment]
$$merge$$

**Pull request merged**